### PR TITLE
Resolves #95: Added Index and UniqueIndex support into MongoModel.

### LINKED ISSUE #95: Add indexes to MongoModel in correctness
So far, `MongoModel`, in correctness code, is silently ignoring indexes. As indexes are not important to verify the correctness of Document Layer. But, some indexes could have an impact on the functionality. We are doing relevant work for unique indexes at #67.

There are other cases like this. We are having quite a few correctness failures due to Multi index limitations, as we don't allow compound indexes on arrays that can create more than 1000 index keys for a document. We should generate errors from MongoModel for this kind of failures to be consistent.

Blocking #67

### PR DESCRIPTION
**Pull request by dongxinEric**

### CODE REVIEW

File: test/correctness/document-correctness.py [hunk 1]
```diff
@@ -300,40 +334,38 @@ def one_iteration(collection1, collection2, ns, seed):
 
         if indexes_first:
             for i in indexes:
-                transactional_shim.ensure_index(collection1, i)
-                transactional_shim.ensure_index(collection2, i)
-
+                # uniqueIndex = gen.global_prng.choice([True, False])
+                uniqueIndex = False
```
Comment (apkar) [inline test/correctness/document-correctness.py]: Good its disabled now. We have a tracking issue #106 once it's done we can reenable this. When we do enable make sure we don't enable for 50% cases. That way unique indexes will cloud everything else. Make it something like 5-10%.

File: test/correctness/mongo_model.py [hunk 2]
```diff
@@ -447,13 +447,17 @@ def __init__(self, options, collectionname=''):
...
         }
+        self.indexes = []
 
     def remove(self):
         self.data = dict()
 
     def _insert(self, doc):
         if '_id' not in doc:
             doc['_id'] = gen.random_object_id()
+        tmp = self.data.values() + [doc]
```
Comment (apkar) [inline test/correctness/mongo_model.py]: We should add a similar code block for updates.

Comment (dongxinEric) [reply test/correctness/mongo_model.py]: Done

**Comment by dongxinEric** [comment]
This would help #47. The new MongoModel won't do the key length check though, cause it's tricky to get the exact KVS byte size of the key, so it simply ignore this error.

**Comment by dongxinEric** [comment]
This would resolve #95

**Pull request merged**